src/listpack.c

```diff
@@ -1206,14 +1206,23 @@ int lpValidateIntegrity(unsigned char *lp, size_t size, int deep,
     return 1;
 }
 
+/* Compare entry pointer to by 'p' with string 's' of length 'slen'.
+ * Return 1 if equal. */
 unsigned int lpCompare(unsigned char *p, unsigned char *s, uint32_t slen) {
-    unsigned char buf[LP_INTBUF_SIZE];
     unsigned char *value;
     int64_t sz;
-
     if (p[0] == LP_EOF) return 0;
-    value = lpGet(p, &sz, buf);
-    return (slen == sz) && memcmp(value,s,slen) == 0;
+
+    value = lpGet(p, &sz, NULL);
+    if (value) {
+        return (slen == sz) && memcmp(value,s,slen) == 0;
+    } else {
+        int64_t sval;
+        if (lpStringToInt64((const char*)s, slen, &sval))
+            return sz == sval;
+    }
+
```

Comment:
so converting a string to int is faster than converting an int to string?
maybe you should state that in a comment, so that someone won't try to "simplify" this code back..

Reply:
Yes, and `memcmp` is also slower than comparing integer directly.

Reply:
Done with the comment.
And change the `snprintf` in lpGet to `ll2string`, I'm not sure why not use ll2string (maybe the original listpack repository isn't here).

Reply:
yes, that's probably it.